src/main/core.js

```diff
@@ -66,41 +68,108 @@ function coreFormat(text, opts, addAlignmentSize) {
   const ast = parsed.ast;
   text = parsed.text;
 
-  let cursorOffset;
   if (opts.cursorOffset >= 0) {
-    const cursorNodeAndParents = rangeUtil.findNodeAtOffset(
-      ast,
-      opts.cursorOffset,
-      opts
-    );
-    const cursorNode = cursorNodeAndParents.node;
-    if (cursorNode) {
-      cursorOffset = opts.cursorOffset - opts.locStart(cursorNode);
-      opts.cursorNode = cursorNode;
+    const nodeResult = rangeUtil.findNodeAtOffset(ast, opts.cursorOffset, opts);
+    if (nodeResult && nodeResult.node) {
+      opts.cursorNode = nodeResult.node;
     }
   }
 
   const astComments = attachComments(text, ast, opts);
   const doc = printAstToDoc(ast, opts, addAlignmentSize);
   opts.newLine = guessLineEnding(text);
 
-  const result = printDocToString(doc, opts);
-  const formatted = result.formatted;
-  const cursorOffsetResult = result.cursor;
+  let result = printDocToString(doc, opts);
+
   ensureAllCommentsPrinted(astComments);
   // Remove extra leading indentation as well as the added indentation after last newline
   if (addAlignmentSize > 0) {
-    return { formatted: formatted.trim() + opts.newLine };
+    const trimmed = result.formatted.trim();
+    const cursorNodeStart =
+      result.cursorNodeStart === undefined
+        ? undefined
+        : result.cursorNodeStart - result.formatted.indexOf(trimmed);
+    result = Object.assign({}, result, {
+      formatted: trimmed + opts.newLine,
+      cursorNodeStart
+    });
```

Comment:
You can change the `result` object here

Reply:
Removing `result =` and `{},` should have that effect.